interactions/client.py

```diff
@@ -631,34 +666,61 @@ def remove(self, name: str, package: Optional[str] = None) -> None:
         :param package?: The package of the extension.
         :type package: Optional[str]
         """
-        _name: str = resolve_name(name, package)
-        module = self._extensions.get(_name)
+        try:
+            _name: str = resolve_name(name, package)
+        except AttributeError:
+            _name = name
+
+        extension = self._extensions.get(_name)
 
-        if module not in self._extensions:
+        if _name not in self._extensions:
             log.error(f"Extension {name} has not been loaded before. Skipping.")
+            return
+
+        try:
+            extension.teardown()  # made for Extension, usable by others
+        except AttributeError:
+            pass
+
+        if isinstance(extension, ModuleType):  # loaded as a module
+            for ext_name, ext in getmembers(
+                extension, lambda x: isinstance(x, type) and issubclass(x, Extension)
+            ):
+                self.remove(ext_name)
+
+            del sys.modules[_name]
 
-        log.debug(f"Removed extension {name}.")
-        del sys.modules[_name]
         del self._extensions[_name]
```

Comment:
```suggestion
        del sys.modules[_name]
        del self._extensions[_name]
```

Reply:
That causes an error when unloading something that isn't a module, e.x. an Extension